packages/jaeger-ui/src/components/TracePage/TraceStatistics/tableValues.tsx

```diff
@@ -45,21 +44,47 @@ function getChildOfSpans(parentID: string, allSpans: Span[]): Span[] {
 
 function computeSelfTime(span: Span, allSpans: Span[]): number {
   if (!span.hasChildren) return span.duration;
-  // We want to represent spans as half-open intervals like [startTime, startTime + duration).
-  // This way the subtraction preserves the right boundaries. However, DRange treats all
-  // intervals as inclusive. For example,
-  //       range(1, 10).subtract(4, 8) => range([1, 3], [9-10])
-  //       length=(3-1)+(10-9)=2+1=3
-  // In other words, we took an interval of length=10-1=9 and subtracted length=8-4=4.
-  // We should've ended up with length 9-4=5, but we got 3.
-  // To work around that, we multiply start/end times by 10 and subtract one from the end.
-  // So instead of [1-10] we get [10-99]. This makes the intervals work like half-open.
-  const spanRange = new DRange(10 * span.startTime, 10 * (span.startTime + span.duration) - 1);
-  const children = getChildOfSpans(span.spanID, allSpans);
-  children.forEach(child => {
-    spanRange.subtract(10 * child.startTime, 10 * (child.startTime + child.duration) - 1);
-  });
-  return Math.round(spanRange.length / 10);
+
+  let spanSelfTime = span.duration;
+  let previousSpanEndTime = span.startTime;
+
+  const children = getChildOfSpans(span.spanID, allSpans).sort((a, b) => a.startTime - b.startTime);
+
+  const parentSpanEndTime = span.startTime + span.duration;
+
+  for (let index = 0; index < children.length; index++) {
+    const child = children[index];
+
+    const spanEndTime = child.startTime + child.duration;
+    const spanStartsAfterParentEnded = child.startTime > parentSpanEndTime;
+    const spanEndsBeforePreviousSpan = spanEndTime < previousSpanEndTime;
```

Comment:
the naming is confusing. Can we stick with `parentXyz` and `childXyz` to make it easier to follow?

Reply:
IMO these names are accurate.

```
parent |..................|
child    |.......|
child     |.....|                      - spanEndsBeforePreviousSpan is true, this span is skipped
child                         |......| - spanStartsAfterParentEnded is true, this span is skipped
```

Reply:
current naming is just `spanXyz`, which is confusing because there are two spans involved in every step and it's not clear which one is referenced by `spanXyz` name

Reply:
I think I misunderstood your initial comment, this is a general issue for the whole function, not just the lines you commented on? I see now what that I used `span` in the parameter of the function and `span` for children.

Can you confirm that these names are less confusing? Or if you have other suggestion it would be helpful if you could provide examples.
* `previousSpanEndTime` => `previousChildEndTime`
* `spanEndTime` => `childEndTime`
* `spanEndsBeforePreviousSpan` => `childEndsBeforePreviousChild` 
* `spanStartsAfterParentEnded` => `childStartsAfterParentEnded` 
* `spanStartsBeforePreviousSpanEnds` => `childStartBeforePreviousChildEnds`
* `spanSelfTime` => `parentSelfTime`?

Reply:
Yes that's better